doc/faq/a05-board-generator.rst

```diff
@@ -0,0 +1,80 @@
+:orphan:
+
+Board generator
+---------------
+
+The board generator is a python script originally intended to ease the
+Arduino IDE's `boards.txt` configuration file about the multitude of
+available board, especially when common data have to be updated for all of
+them.
+
+This script is also used to manage uncommon options that will not find their
+place into the IDE menu.
+
+-  `How can I run the script ? <#how-can-i-run-the-script>`__
+-  `What can I do with it ? <#what-can-i-do-with-it>`__
+-  `When do I need to update it ? <#when-do-i-need-to-mess-with-it>`__
+-  `Why is my pull-request failing continuous-integration ? <#why-is-my-pull-request-failing-continuous-integration>`__
+
+How can I run the script ?
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Python 2 needs to be installed on your system.
+
+The script is located in the ``tools`` subdirectory of the core's root installation.
+It needs to be run from the root directory,
+
+::
+
+    $ tools/boards.txt.py
+
+::
+
+    C:\...> tools\boards.txt.py
+
+Running without parameters will show the command line help.  They are
+generally self-explanatory.
+
+
+What can I do with it ?
+~~~~~~~~~~~~~~~~~~~~~~~
+
+As of today you can:
+
+* in the IDE: change the default serial programming speed of any board
+
+* in the IDE: add new serial programming speed
+
+* increase available flash space by disabling floats in ``*printf`` functions
+
+* enable WPS which is now disabled by default (at the cost of a smaller heap by ~4KB)
```

Comment:
in the IDE?

Reply:
all of these are in the IDE, right?

Reply:
float/printf and WPS are not in IDE. The board generator script is currently the only way to change these.